sql/src/test/java/io/crate/operation/scalar/TypeInferenceTest.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar;
+
+import org.junit.Test;
+
+public class TypeInferenceTest extends AbstractScalarFunctionsTest {
+
+    @Test
+    public void testCoalesce() throws Exception {
+        assertEvaluate("coalesce(null, 1::integer, 2::long, 3.0)", 1.0);
```

Comment:
Could we also add a test with a column that shows that integer casts a no longer necessary for numeric literals?

Reply:
I've added a test in `FuncParamsTest`. Adding another one to test this end-to-end.